pkg/components/beyla.go

```diff
@@ -70,18 +70,17 @@ func RunBeyla(ctx context.Context, cfg *beyla.Config) error {
 func setupAppO11y(ctx context.Context, ctxInfo *global.ContextInfo, config *beyla.Config) error {
 	slog.Info("starting Beyla in Application Observability mode")
 
-	wg := sync.WaitGroup{}
-	defer wg.Wait()
-
 	instr := appolly.New(ctx, ctxInfo, config)
-	if err := instr.FindAndInstrument(&wg); err != nil {
```

Comment:
This change avoids sharing the ownership of a critical synchronization mechanism between different components.

The usual convention in Go is expose a simpler synchronization mechanism, such as returning a close channel.